pkg/k8s/client.go

```diff
@@ -63,26 +63,33 @@ func createConfig(endpoint, kubeCfgPath string) (*rest.Config, error) {
 		if err != nil {
 			return nil, err
 		}
-		config.UserAgent = userAgent
+		setConfig(config, userAgent, qps, burst)
 		return config, nil
 	}
 
 	config := &rest.Config{Host: endpoint, UserAgent: userAgent}
+	setConfig(config, userAgent, qps, burst)
 	err := rest.SetKubernetesDefaults(config)
 
 	return config, err
 }
 
+func setConfig(config *rest.Config, userAgent string, qps float32, burst int) {
+	config.UserAgent = userAgent
```

Comment:
Don't blindly set all fields, check their defaults first, if they are different then defaults then set the field. For example:

```golang
if qps != 0.0 {
     config.QPS = qps
}
```